src/basic/log.h

```diff
@@ -367,3 +368,41 @@ int log_syntax_invalid_utf8_internal(
 #define DEBUG_LOGGING _unlikely_(log_get_max_level() >= LOG_DEBUG)
 
 void log_setup(void);
+
+typedef struct ErrorRateLimit {
+        int error;
+        RateLimit ratelimit;
+} ErrorRateLimit;
+
+void log_dropped_errors(size_t num_dropped_errors, int level);
+
+bool error_is_ratelimited(int error, ErrorRateLimit *error_ratelimit);
```

Comment:
we try to prefix all log related stuff with `log_`, hence this should carry a similar prefix